src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs

```diff
@@ -185,7 +184,75 @@ public override void WriteValue(object value)
         {
             ArgumentNullException.ThrowIfNull(value);
 
-            WriteString(XmlUntypedConverter.Untyped.ToString(value, _resolver));
+            Type sourceType = value.GetType();
```

Comment:
This seems to be adding a lot of branching where it previously didn't exist. Could this be done differently? I'm not super familiar with this code, but one thought might be to expose a `TryFormat` method to `XmlUntypedConverter.Untyped`.

Reply:
The code in XmlUntypedConverter.ToString(object) is already similar:
    ```cs
public override string ToString(object value, IXmlNamespaceResolver? nsResolver)
    {
        ArgumentNullException.ThrowIfNull(value);

        Type sourceType = value.GetType();

        if (sourceType == BooleanType) return XmlConvert.ToString((bool)value);
        if (sourceType == ByteType) return XmlConvert.ToString((byte)value);
        if (sourceType == ByteArrayType) return Base64BinaryToString((byte[])value);
        if (sourceType == DateTimeType) return DateTimeToString((DateTime)value);
        if (sourceType == DateTimeOffsetType) return DateTimeOffsetToString((DateTimeOffset)value);
        if (sourceType == DecimalType) return XmlConvert.ToString((decimal)value);
        if (sourceType == DoubleType) return XmlConvert.ToString((double)value);
        if (sourceType == Int16Type) return XmlConvert.ToString((short)value);
        if (sourceType == Int32Type) return XmlConvert.ToString((int)value);
        if (sourceType == Int64Type) return XmlConvert.ToString((long)value);
        if (sourceType == SByteType) return XmlConvert.ToString((sbyte)value);
        if (sourceType == SingleType) return XmlConvert.ToString((float)value);
        if (sourceType == StringType) return ((string)value);
        if (sourceType == TimeSpanType) return DurationToString((TimeSpan)value);
        if (sourceType == UInt16Type) return XmlConvert.ToString((ushort)value);
        if (sourceType == UInt32Type) return XmlConvert.ToString((uint)value);
        if (sourceType == UInt64Type) return XmlConvert.ToString((ulong)value);
        if (IsDerivedFrom(sourceType, UriType)) return AnyUriToString((Uri)value);
        if (sourceType == XmlAtomicValueType) return ((string)((XmlAtomicValue)value).ValueAs(StringType, nsResolver));
        if (IsDerivedFrom(sourceType, XmlQualifiedNameType)) return QNameToString((XmlQualifiedName)value, nsResolver);

        return (string)ChangeTypeWildcardDestination(value, StringType, nsResolver);
    }
```